cql-engine/src/main/java/org/opencds/cqf/cql/elm/execution/ExpressionDefEvaluator.java

```diff
@@ -14,10 +14,13 @@ protected Object internalEvaluate(Context context) {
         }
         try {
             if (context.isExpressionCachingEnabled() && context.isExpressionInCache(this.getName())) {
+                context.setCurrentExpression(this.getName());
                 return context.getExpressionResultFromCache(this.getName());
             }
 
+            context.setCurrentExpression(this.getName());
             Object result = this.getExpression().evaluate(context);
```

Comment:
This should be protected with a try..finally, yes?

Reply:
Isn't it?

Reply:
No. It's true that it's inside a larger try..finally, but that is protecting the evaluation context. In line 21, we are pushing the current expression onto a stack. In line 23, we are popping it. If an error occurs in line 22, line 23 will never execute, and we'll be leaving the stack imbalanced.